components/schedule-editor/src/ScheduleEditor.svelte

```diff
@@ -293,6 +303,29 @@
   let slots_to_book: TimeSlot;
   let mainElementWidth: number;
   $: showPreview = mainElementWidth > 600;
+
+  //#region custom fields
+  const customFieldKeys = ["title", "description", "type", "required"];
+  const emptyCustomField: CustomField = {
+    title: "",
+    description: "",
+    type: "text",
+    required: false,
+  };
+
+  let newFieldTitleElement: HTMLElement;
+  let newCustomField: CustomField = { ...emptyCustomField };
```

Comment:
I think I could probably also Object.freeze to do this, but I figured a {... clone} would do the trick.